bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/item/ItemResource.java

```diff
@@ -367,6 +368,90 @@ public Response removeTag(@PathParam("itemname") String itemname, @PathParam("ta
         return Response.ok().build();
     }
 
+    /**
+     * Create or Update an item by supplying an item bean.
+     * @param itemname
+     * @param item the item bean.
+     * @return
+     */
+    @PUT
+    @Path("/{itemname: [a-zA-Z_0-9]*}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createOrUpdateItem(@PathParam("itemname") String itemname, ItemBean item) {
```

Comment:
Please only remove the old "createOrUpdate" methods. The Paper UI does not use any of these methods any more, but I think it makes sense to have them for other UIs. For example if you implement a UI component that only allows you to add tags to an Item, it´s easier do it with the other resource methods. So, if you agree I would suggest to keep the other methods.